src/Maybe.js

```diff
@@ -29,11 +35,19 @@ Maybe.of = Maybe.Just;
 Maybe.prototype.of = Maybe.Just;
 
 Maybe.isJust = function(x) {
-  return x instanceof _Just;
+  if (typeof x.isJust !== 'boolean') {
```

Comment:
We can still keep the types signature to specify that only `Maybe` instances are accepted. It's just a matter of whether we throw in the implementation if something other than a `Maybe` instance is given.

If we're to add type checking to functions, we will need to approach it more holistically such as the suggestion in #89.